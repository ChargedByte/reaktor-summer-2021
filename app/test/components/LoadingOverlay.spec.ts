import { createLocalVue, mount } from '@vue/test-utils'

import Vuetify from 'vuetify'

import LoadingOverlay from '~/components/LoadingOverlay.vue'

describe('LoadingOverlay', () => {
  const localVue = createLocalVue()
  let vuetify: Vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  test('is a Vue instance', () => {
    const wrapper = mount(LoadingOverlay, { localVue, vuetify })
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly: on', () => {
    const wrapper = mount(LoadingOverlay, {
      localVue,
      vuetify,
      propsData: { value: true },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('renders correctly: off', () => {
    const wrapper = mount(LoadingOverlay, {
      localVue,
      vuetify,
      propsData: { value: false },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
