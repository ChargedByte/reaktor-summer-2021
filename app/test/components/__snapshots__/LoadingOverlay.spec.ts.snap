// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingOverlay renders correctly: off 1`] = `
"<div class=\\"v-overlay theme--dark\\" style=\\"z-index: 5;\\">
  <div class=\\"v-overlay__scrim\\" style=\\"opacity: 0; background-color: rgb(33, 33, 33); border-color: #212121;\\"></div>
</div>"
`;

exports[`LoadingOverlay renders correctly: on 1`] = `
"<div class=\\"v-overlay v-overlay--active theme--dark\\" style=\\"z-index: 5;\\">
  <div class=\\"v-overlay__scrim\\" style=\\"opacity: 1; background-color: rgb(33, 33, 33); border-color: #212121;\\"></div>
  <div class=\\"v-overlay__content\\">
    <div role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" class=\\"v-progress-circular v-progress-circular--visible v-progress-circular--indeterminate primary--text\\" style=\\"height: 64px; width: 64px;\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"21.333333333333332 21.333333333333332 42.666666666666664 42.666666666666664\\" style=\\"transform: rotate(0deg);\\">
        <circle fill=\\"transparent\\" cx=\\"42.666666666666664\\" cy=\\"42.666666666666664\\" r=\\"20\\" stroke-width=\\"2.6666666666666665\\" stroke-dasharray=\\"125.664\\" stroke-dashoffset=\\"125.66370614359172px\\" class=\\"v-progress-circular__overlay\\"></circle>
      </svg>
      <div class=\\"v-progress-circular__info\\"></div>
    </div>
  </div>
</div>"
`;
